astroquery/query.py

```diff
@@ -185,11 +185,19 @@ class BaseVOQuery:
     """
     def __init__(self):
         super().__init__()
-        self._session = requests.Session()
-        self._session.headers['User-Agent'] = (
-            f"astroquery/{version.version} pyVO/{pyvo.__version__} Python/{platform.python_version()} "
-            f"({platform.system()}) "
-            f"{self._session.headers['User-Agent']}")
+        if not hasattr(self, '_session'):
+            # We don't want to override another, e.g. already authenticated session from another baseclass
+            self._session = requests.Session()
+
+        user_agents = self._session.headers['User-Agent'].split()
+        if 'astroquery' in user_agents[0]:
+            if 'pyVO' not in user_agents[1]:
+                user_agents[0] = f"astroquery/{version.version} pyVO/{pyvo.__version__}"
+        else:
+            user_agents = [f"astroquery/{version.version} pyVO/{pyvo.__version__} "
+                           f"Python/{platform.python_version()} ({platform.system()})"] + user_agents
+
```

Comment:
This is not potentially repeating the Python version?

Reply:
Potentially yes, but it's already the case in the current `main`, I just made sure that we don't do this if we (==astroquery or pyvo) already modified the user-agent. The one corner case that I haven't covered if the session is somehow made in pyvo, and thus only have the pyvo version listed, but I don't think that will ever be the case here.

Reply:
`PyVO` might be used to create auth sessions if that is going to be standardized eventually. It already has `authsession.AuthSession` although that is a bit outdated.

Reply:
If I see correctly, then authsession.AuthSession doesn't modify the headers. Though yes, I suppose the above logic should better be fixed in case it would modify it.

Reply:
The latest commit should cover the case when PyVO already modified the user-agent. 

Some other tools might have already added the python and platform specifics, too, but I think we should just roll with that and add our own, too, and thus duplicate if that's the case.